src/Files.App/Actions/FileSystem/Transfer/BaseTransferItemAction.cs

```
@@ -0,0 +1,129 @@
+							items.Add(await ftpItem.ToStorageItem());
+					}
+					else
+					{
+						if (type is DataPackageOperation.Move)
+						{
+							// Dim opacities accordingly
+							await MainWindow.Instance.DispatcherQueue.EnqueueOrInvokeAsync(() =>
+							{
+								listedItem.Opacity = Constants.UI.DimItemOpacity;
+							});
+						}
+
+						FilesystemResult? result =
+							listedItem.PrimaryItemAttribute == StorageItemTypes.File || listedItem is ZipItem
+								? await ContentPageContext.ShellPage.ShellViewModel.GetFileFromPathAsync(listedItem.ItemPath).OnSuccess(t => items.Add(t))
+								: await ContentPageContext.ShellPage.ShellViewModel.GetFolderFromPathAsync(listedItem.ItemPath).OnSuccess(t => items.Add(t));
+
+						if (!result)
+							throw new IOException($"Failed to process {listedItem.ItemPath} in cutting/copying to the clipboard.", (int)result.ErrorCode);
+					}
+				},
+				10,
+				statusCenterItem?.CancellationToken ?? default);
+
+				var standardObjectsOnly = items.All(x => x is StorageFile or StorageFolder or SystemStorageFile or SystemStorageFolder);
+				if (standardObjectsOnly)
+					items = new ConcurrentBag<IStorageItem>(await items.ToStandardStorageItemsAsync());
+
+				if (items.IsEmpty)
+					return;
+
+				dataPackage.Properties.PackageFamilyName = Windows.ApplicationModel.Package.Current.Id.FamilyName;
+				dataPackage.SetStorageItems(items, false);
+
+				Clipboard.SetContent(dataPackage);
+			}
+			catch (Exception ex)
+			{
+				dataPackage = default;
+
+				if ((FileSystemStatusCode)ex.HResult is FileSystemStatusCode.Unauthorized)
+				{
+					string[] filePaths = ContentPageContext.SelectedItems.Select(x => x.ItemPath).ToArray();
+					await FileOperationsHelpers.SetClipboard(filePaths, type);
+
+					return;
+				}
+
+				// Reset cut mode
+				ContentPageContext.ShellPage.SlimContentPage.ItemManipulationModel.RefreshItemsOpacity();
+
+				return;
```

@hishitetsu I’m wondering if the app should report the produced exception since this try-catch may throw IOException (and others in WinRT APIs) before returning.

---

If the exception occurs for unexpected reasons, it should be reported. But if you know why the exception occurs, for example, it occurs by a known user operation, it would be better not to report it.

---

Done.